File: src/main/java/com/example/chatService/service/ParticipantEventPublisher.java
package com.example.chatService.service;

import com.example.chatService.dto.ParticipantDto;

public interface ParticipantEventPublisher {
    void broadcastJoin(String roomId, ParticipantDto participant);

    void broadcastLeave(String roomId, ParticipantDto participant);

    void broadcastLeave(String roomId, ParticipantDto participant, String reason);

    void broadcastOwnerChanged(String roomId, String newOwnerId);
}

File: src/main/java/com/example/chatService/service/ParticipantEventPublisher.kt
package com.example.chatService.service

import com.example.chatService.dto.ParticipantDto

interface ParticipantEventPublisher {
    fun broadcastJoin(roomId: String, participant: ParticipantDto)

    fun broadcastLeave(roomId: String, participant: ParticipantDto)

    fun broadcastLeave(roomId: String, participant: ParticipantDto, reason: String?)

    fun broadcastOwnerChanged(roomId: String, newOwnerId: String)
}
